### ISSUE TITLE
Consider removing manual modularization

### ISSUE THREAD

**Issue by shamrin** [sql-js/sql.js#438]
The generated code in `sql-wasm.js` [mentions][1] Emscripten issue: emscripten-core/emscripten#5820.

This issue was fixed in PR emscripten-core/emscripten#10697 and [released as version 13.9.16](https://github.com/emscripten-core/emscripten/blob/master/ChangeLog.md#13916-05152020) in May 2020.

I think removing manual modularization will simplify sql.js build process.

[1]: https://github.com/sql-js/sql.js/blob/master/src/shell-pre.js

**Comment by lovasoa** [comment]
Yes, this could be nice, and simplify our old build process. Are you interested in making a PR ?

**Comment by Taytay** [comment]
I'm thrilled to hear that this was fixed in Emscripten directly. I've been out of this project for a while, but am dipping my toes back in and can take a look if no one beats me to it.

**Comment by rhashimoto** [comment]
I don't know how the bug was originally triggered, but I tried adding `-s MODULARIZE=1` to the emcc flags, changed the output filename to end with .mjs (to output an ES6 module which I prefer), and removed the custom bundling.

Then to load the module in my page I did:
```
import initSqlJs from '../dist/sql-idb-wasm.mjs';
initSqlJs().then(async SQL => {
  ...
```

It seems to work. I'm using emsdk 2.0.16. Again, I don't know if I'm exercising the problem path.

**Comment by rhashimoto** [comment]
Hi @Taytay, do you have a time frame for working on this? And would it be possible to fold in #284?

@lovasoa has greenlit elements of #447, and although better modularization isn't a necessity it would certainly be welcomed to streamline development. If I'm not mistaken it will also reduce the .js size by a few percent since the ad hoc wrapper isn't minified, so slightly better load time as well.
